### ISSUE TITLE
Problems when library is compiled locally

### ISSUE THREAD

**Issue by blelem** [microsoft/vs-streamjsonrpc#360]
TLDR: Is there an eminent nuget release planned?

My project uses the "Add support for custom request IDs (including strings)" functionality which isn't yet available in the current Nuget package. I am having issues building the library, and I expect that a Nuget release would fix all my issues. 

In case a new release isn't yet on the horizon, here are the issues I encountered when building. 
 
I had a successful setup before (a few months ago), where I could build the library and use it in my app, but I can't figure out how to get a functional app from the current master branch. 

Here is my setup. I have a dotnet core 3.0 application. I included StreamJsonRpc project as one of the projects of the solution.

When I build (dotnet publish -c release), I get a 
         `/home/lemieber/.nuget/packages/xlifftasks/1.0.0-beta.19253.2/build/XliffTasks.targets(92,5): error : 'xlf/Resources.cs.cs.xlf' for 'Resources.cs.resx' does not exist. Run `msbuild /t:UpdateXlf` to update .xlf files or set UpdateXlfOnBuild=true to update them on every build, but note that it is strongly discouraged to set UpdateXlfOnBuild=true in official/CI build environments as they should not modify source code during the build. [/home/lemieber/unmannedAPT.streamjsonrpc/src/StreamJsonRpc/StreamJsonRpc.csproj]`

I didn't quite knew how to address that, but removing XliffTasks from StreamJsonRpc.csproj seems to work. With that fixed, I got a little further, until compilation complains about dotnet 3.0:
` error NETSDK1005: Assets file '/home/lemieber/unmannedAPT.streamjsonrpc/obj/StreamJsonRpc/project.assets.json' doesn't have a target for '.NETCoreApp,Version=v3.0'.`
I've added netcoreapp3.0 to the StreamJsonRpc.csproj.  At this stage, my app works fine if I deploy it by copying the  whole build directory.

But if I try to deploy via a debian (.deb) package, I get a runtime error:
`Could not load file or assembly 'Nerdbank.Streams, Version=2.4.0.0, Culture=neutral, PublicKeyToken=cac503e1823ce71c'. The located assembly's manifest definition does not match the assembly reference. (0x80131040)
File name: 'Nerdbank.Streams, Version=2.4.0.0, Culture=neutral, PublicKeyToken=cac503e1823ce71c'` 

The Nerdbank.Streams.dll is inside the deb package, I really don't have a clue of what could cause this problem. Same deployment method works, when I use an older StreamJsonRpc, based on the commit fad30141: Freshen up master's optprof data too.

None of these issues are showstopper, I can keep using my old patched release of StreamJsonRpc. But if any of these issues have an obvious fix, It would be nice to hear what I did wrong.

Thanks!

**Comment by AArnott** [comment]
Wow, that sounds horrific. Out of curiosity can you build the library when it's in its own repo? I wonder if it's a unique problem to building within your solution or if it's on your machine due to some other configuration.

Anyway, yes a 2.3-alpha version is expected to be pushed to nuget.org very soon. Perhaps later today. In a few minutes the package will appear on our public feed [over here](https://dev.azure.com/azure-public/vside/_packaging?_a=feed&feed=vssdk) as well, so even if it's not on nuget.org right away, you can get it from there.

**Comment by blelem** [comment]
Superb timing, great! I'll report how well the 2.3-alpha does in my environment. 

> Out of curiosity can you build the library when it's in its own repo?

If I try to build the project, with the dotnet sdk, I get the following error

    dotnet build

`/usr/share/dotnet/sdk/3.0.100/Microsoft.Common.CurrentVersion.targets(3056,5): error MSB3577: Two output file names resolved to the same output path: "/home/lemieber/unmannedAPT.streamjsonrpc/obj/StreamJsonRpc/Debug/netstandard2.0/StreamJsonRpc.Resources.cs.resources" [/home/lemieber/unmannedAPT.streamjsonrpc/src/StreamJsonRpc/StreamJsonRpc.csproj]`

If I specify the framework

    dotnet build -f  netstandard2.0

`/usr/share/dotnet/sdk/3.0.100/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(234,5): error NETSDK1005: Assets file '/home/lemieber/unmannedAPT.streamjsonrpc/obj/Benchmarks/project.assets.json' doesn't have a target for '.NETStandard,Version=v2.0'. Ensure that restore has run and that you have included 'netstandard2.0' in the TargetFrameworks for your project. [/home/lemieber/unmannedAPT.streamjsonrpc/src/Benchmarks/Benchmarks.csproj]`

**Comment by blelem** [comment]
I tried my project with the 2.3-alpha build, and it seems to work perfectly. Again big big thanks!

For most of the day, I had the same issue with the wrong version of Nerdbank.Streams.dll. It turned out that the command `dotnet clean` doesn't quite work on my project, most of the bin and obj folders were not deleted, and the build picked an old version of Nerdbank from one of those non-deleted temporary build directory. Doing a manual clean did the trick.

**Comment by AArnott** [comment]
I wonder if a dirty directory explains the problem you had with _building_ this repo on your machine too. I've seen errors like yours before, but not in this repo, especially when it's clean.

**Comment by blelem** [comment]
I get a very different message now, when trying to build the repo. I emptied my nuget cache, cloned the project in another directory.

 in folder ~/vs-streamjsonrpc/src/
 I run:
     
    dotnet build 

`/home/lemieber/.nuget/packages/nerdbank.gitversioning/3.0.26/build/Nerdbank.GitVersioning.targets(63,5): error MSB4062: The "Nerdbank.GitVersioning.Tasks.GetBuildVersion" task could not be loaded from the assembly /home/lemieber/.nuget/packages/nerdbank.gitversioning/3.0.26/build/MSBuildCore/Nerdbank.GitVersioning.Tasks.dll. Assembly with same name is already loaded Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [/home/lemieber/vs-streamjsonrpc/src/StreamJsonRpc.Tests.ExternalAssembly/StreamJsonRpc.Tests.ExternalAssembly.csproj]`

**Comment by AArnott** [comment]
> Assembly with same name is already loaded

Whenever you see this, run `dotnet build-server shutdown` and try again.

**Comment by blelem** [comment]
It's now successfully building. Woot!
TIL about `build-server shutdown` command.
